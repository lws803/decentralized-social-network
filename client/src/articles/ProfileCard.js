import React from "react";
import PropTypes from "prop-types";

import moment from "moment";
import styled from "styled-components";

import Bio from "../profile/Bio";
import ProfileImage from "../profile/ProfileImage";

const Card = props => {
  return (
    <ProfileCardContainer>
      <ProfileImage profilePhoto={props.authorPhoto} width={54} height={54} />
      <ArticleDetails>
        <AuthorName onClick={props.onProfileClick}>
          {props.authorName}
        </AuthorName>
        <TimeDate>
          {props.dateCreated
            ? props.dateCreated.local().format("DD MMM, YYYY")
            : undefined}
        </TimeDate>
      </ArticleDetails>
      <ButtonContainer>
        <button onClick={props.onFollowClick} disabled>
          Follow
        </button>
      </ButtonContainer>
    </ProfileCardContainer>
  );
};

Card.propTypes = {
  authorPhoto: PropTypes.string,
  authorName: PropTypes.string,
  dateCreated: PropTypes.instanceOf(moment),
  onFollowClick: PropTypes.func,
  onProfileClick: PropTypes.func,
};

const ProfileCardContainer = styled.div`
  width: 280px;
  height: 58px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const ArticleDetails = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 10px;
`;

const TimeDate = styled.div`
  margin-top: 5px;
  font-weight: 200;
`;

const ButtonContainer = styled.div`
  height: 100%;
  padding-top: 10px;
`;

const AuthorName = styled.div`
  cursor: pointer;
  :hover {
    text-decoration: underline;
  }
`;

const LargeCard = props => {
  return (
    <LargeCardContainer>
      <ProfileImage profilePhoto={props.authorPhoto} width={127} height={127} />
      <ProfileDetails>
        <LargeAuthorName onClick={props.onProfileClick}>
          {props.authorName}
        </LargeAuthorName>
        <BioContainer>
          <Bio content={props.bio} />
        </BioContainer>
      </ProfileDetails>
    </LargeCardContainer>
  );
};

LargeCard.propTypes = {
  authorPhoto: PropTypes.string,
  authorName: PropTypes.string,
  bio: PropTypes.string,
  onProfileClick: PropTypes.func,
};

const LargeCardContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const ProfileDetails = styled.div`
  display: flex;
  flex-direction: column;
  margin-left: 10px;
`;

const LargeAuthorName = styled.div`
  font-weight: bold;
  font-size: 20px;
  cursor: pointer;
  :hover {
    text-decoration: underline;
  }
`;

const BioContainer = styled.div`
  margin-top: 10px;
  font-size: 15px;
`;

export { Card, LargeCard };
