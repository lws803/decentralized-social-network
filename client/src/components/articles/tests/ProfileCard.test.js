import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from "enzyme-to-json";
import moment from "moment";

import { Card, LargeCard } from "../ProfileCard";
import ProfilePhoto from "../../../res/7874219.jpeg";

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Card component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Card
        authorName="lws803"
        authorPhoto={ProfilePhoto}
        dateCreated={moment()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Large Card component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <LargeCard
        authorName="lws803"
        authorPhoto={ProfilePhoto}
        bio="The quick brown fox"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
