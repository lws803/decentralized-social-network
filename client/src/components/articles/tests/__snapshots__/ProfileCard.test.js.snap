// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders correctly 1`] = `
<styled.div>
  <ProfileImage
    height={54}
    profilePhoto="7874219.jpeg"
    width={54}
  />
  <styled.div>
    <styled.div>
      lws803
    </styled.div>
    <styled.div>
      30 Aug, 2020
    </styled.div>
  </styled.div>
  <styled.div>
    <button
      disabled={true}
    >
      Follow
    </button>
  </styled.div>
</styled.div>
`;

exports[`Large Card component renders correctly 1`] = `
<styled.div>
  <ProfileImage
    height={127}
    profilePhoto="7874219.jpeg"
    width={127}
  />
  <styled.div>
    <styled.div>
      lws803
    </styled.div>
    <styled.div>
      <Bio
        content="The quick brown fox"
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
